src/serializer/ResidualHeapSerializer.js

```diff
@@ -1163,17 +1173,52 @@ export class ResidualHeapSerializer {
     // TODO #20: add timers
 
     // TODO #21: add event listeners
+
     for (let [moduleId, moduleValue] of this.modules.initializedModules)
       this.requireReturns.set(moduleId, this.serializeValue(moduleValue));
 
+    // Make sure additional functions get serialized.
+    // TODO: make sure those functions get skipped over in all relevant places in serialization/optimization
+    // We have to do this if we're on the first pass so effects get reversed properly
```

Comment:
This comment perplexes me because of the "if we're" bit. Did you mean "We have to do this on the first pass ..."?